http/http.go

```diff
@@ -18,6 +19,59 @@ import (
 
 var tracer = trace.GlobalTracer
 
+type HTTPClientTracer struct {
+	prefix           string
+	stats            *statsd.Client
+	dnsStartTime     time.Time
+	connectStartTime time.Time
+	connectDoneTime  time.Time
+	// If a connection is reused it won't have this, so set it now for safety
+	writeDoneTime time.Time
+}
+
+func NewHTTPClientTracer(stats *statsd.Client, prefix string) *HTTPClientTracer {
+	return &HTTPClientTracer{
+		prefix:          prefix,
+		stats:           stats,
+		connectDoneTime: time.Now(),
+	}
```

Comment:
Curious why you're not making spans here - this seems like the perfect place for them, and it would lead to far less statsd udp chunder.

Reply:
This is a pretty good idea. The wrinkle is that these methods can be called concurrently, so to keep the "current" span we'd have to also use a mutex such that we had a `currentSpan = trace.ChildSpan(currentSpan)` or whatever.

This feels much better to me. I think i'll add it.

Reply:
On reflection, this feels a bit icky. I'm not sure if it will work right because: 
> ClientTrace is a set of hooks to run at various stages of an outgoing HTTP request. Any particular hook may be nil. Functions may be called concurrently from different goroutines and some may be called after the request has completed or failed.

Like, even these timings are a bit suspect upon reflection. It might be better to just take them out since reused connections will only emit write and response… and that's not a big deal.

Maybe the net effect is just to emit reuse metrics and overall connection time.

Reply:
So one thing you could do is, upon creating the client tracer, make one new parallel span per callback we expose (all parented to the current span), and in each of these marker callbacks, finish that corresponding span. That'll not look like a sequential graph of things, but it'll at least let us *see* in a trace what happened when.